### PR TITLE
Fix test-upload merge general release notes

### PR DESCRIPTION
**Pull request by DinaMeylakh**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue](https://jira-dc.paloaltonetworks.com/browse/CIAC-7940)

## Description
When using the test-upload flow we aggregate release notes changes together when publishing what was changed. Merging release notes with general comments results in their deletion. In this pr I fix the merge logic to include the general release notes.

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Comment by DinaMeylakh** [comment]
TDD: failing unittests commit [4eb63a8](https://github.com/demisto/content/pull/30041/commits/4eb63a835c9853af2f0b48b273b4f3af2810a039) in this [job](https://code.pan.run/xsoar/content/-/jobs/34102821)

**Comment by DinaMeylakh** [comment]
Tested with Elasticsearch pack in commit [f30c889](https://github.com/demisto/content/pull/30041/commits/f30c8898d5e86c048b75e712a6650043360245a6) 
A snippet from the generated `content-descriptor.json`  containing a successful merger:
Screenshot 2023-10-08 at 16 12 23

Link to the [packs-release-notes.md](https://code.pan.run/xsoar/content/-/jobs/34314987/artifacts/file/artifacts/xsoar/packs-release-notes.md) artifact.
Link to the [content-descriptor.json](https://code.pan.run/xsoar/content/-/jobs/34314987/artifacts/external_file/artifacts/xsoar/content-descriptor.json) artifact.

**Comment by DinaMeylakh** [comment]
Tested general release notes merger with Expanse V2 in commit [ab58b5d](https://github.com/demisto/content/pull/30041/commits/ab58b5da48d212fede017f07d4c5fec49cfa9b40)
Snippet from the [packs-release-notes.md](https://code.pan.run/xsoar/content/-/jobs/35265513/artifacts/file/artifacts/xsoar/packs-release-notes.md) containing a successful merger:
Screenshot 2023-10-25 at 18 13 12

File: Utils/release_notes_generator.py [hunk 1]
```diff
@@ -324,6 +369,22 @@ def merge_version_blocks(pack_versions_dict: dict, return_str: bool = True) -> t
...
+        # Handle general pack notes.
+        general_note_sections = PACK_GENERAL_NOTES_REGEX.findall(version_release_notes)
+        if general_note_sections:
+            # Assume only one general notes section at the start of the file.
+            general_note_components = general_note_sections[0]
+            entity_type = ContentType.PACK.value
+            entities_data.setdefault(entity_type, {})
+            # Extract Pack name
+            entity_name = general_note_components[0] or general_note_components[2] or general_note_components[4]
+            entity_name = entity_name.replace('__', '')
```
Comment (ShahafBenYakir) [inline Utils/release_notes_generator.py:382]: Remove

File: Utils/release_notes_generator.py [hunk 2]
```diff
@@ -324,6 +369,22 @@ def merge_version_blocks(pack_versions_dict: dict, return_str: bool = True) -> t
...
+            # release notes of the entity
+            last_place_entity_comment = general_note_components[5] if len(general_note_components) >= 5 else ''
```
Comment (ShahafBenYakir) [inline Utils/release_notes_generator.py:384]: remove

File: Utils/release_notes_generator.py [hunk 3]
```diff
@@ -324,6 +369,22 @@ def merge_version_blocks(pack_versions_dict: dict, return_str: bool = True) -> t
...
+            entity_comment = general_note_components[1] or general_note_components[3] or last_place_entity_comment
```
Comment (ShahafBenYakir) [inline Utils/release_notes_generator.py:385]:
```suggestion
            entity_comment = general_note_components[1] or general_note_components[3]
```

**Comment by DinaMeylakh** [comment]
Tested general release notes merger with RTIR in commit [2fcf903](https://github.com/demisto/content/pull/30041/commits/2fcf903ae413b819e9f025b5e4894d4d2d6b3cb0)
Snippet from the [packs-release-notes.md](https://code.pan.run/xsoar/content/-/jobs/35515203/artifacts/file/artifacts/xsoar/packs-release-notes.md) containing a successful merger:
image

**Comment by DinaMeylakh** [comment]
@ilaner @ShahafBenYakir Please also review the addition of display_name to pack entities in graph https://github.com/demisto/demisto-sdk/pull/3753

**Review by ShahafBenYakir — APPROVED** [review]

**Pull request merged**